src/main/java/com/bookstore/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional(readOnly = true)
+    public ShoppingCartResponseDto getCartForCurrentUser() {
+        User user = getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(user)
+                .orElseGet(() -> createCartForUser(user));
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto addItemToCart(CartItemRequestDto requestDto) {
+        User user = getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(user)
+                .orElseGet(() -> createCartForUser(user));
+
+        Book book = bookRepository.findById(requestDto.bookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with id: " + requestDto.bookId()));
+
+        CartItem existingItem = cart.getCartItems().stream()
+                .filter(ci -> ci.getBook().getId().equals(book.getId()))
+                .findFirst()
+                .orElse(null);
+
+        if (existingItem != null) {
+            existingItem.setQuantity(existingItem.getQuantity() + requestDto.quantity());
+        } else {
+            CartItem newItem = new CartItem();
+            newItem.setBook(book);
+            newItem.setQuantity(requestDto.quantity());
+            newItem.setShoppingCart(cart);
+            cart.getCartItems().add(newItem);
+        }
+
+        shoppingCartRepository.save(cart);
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    @Transactional
+    public CartItemResponseDto updateCartItemQuantity(Long cartItemId,
+                                                      CartItemUpdateRequestDto requestDto) {
+        CartItem item = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Cart item not found with id: " + cartItemId));
+        item.setQuantity(requestDto.getQuantity());
+        cartItemRepository.save(item);
+        return cartItemMapper.toDto(item);
+    }
+
+    @Override
+    @Transactional
+    public void removeCartItem(Long cartItemId) {
+        if (!cartItemRepository.existsById(cartItemId)) {
+            throw new EntityNotFoundException("Cart item not found with id: " + cartItemId);
+        }
+        cartItemRepository.deleteById(cartItemId);
+    }
+
+    private ShoppingCart createCartForUser(User user) {
```

Comment:
create shop cart for user during registration
```suggestion
    @Override
    public void createCartForUser(User user) {
```